iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -0,0 +1,227 @@
+
+                            InitializeClient();
+                        }
+                    }
+                    break;
+            }
+        }
+
+        public async Task RunSampleAsync()
+        {
+            await Task.WhenAll(SendMessagesAsync(), ReceiveMessagesAsync());
+        }
+
+        private void InitializeClient()
+        {
+            // If the client reports Connected status, it is already in operational state.
+            if (connectionStatus != ConnectionStatus.Connected)
+            {
+                lock (_lock)
+                {
+                    _logger.LogDebug($"Attempting to reinitialize the client instance, current status={connectionStatus}");
+
+                    // If the device client instance has been previously initialized, then dispose it.
+                    // The wasConnected variable is required to store if the client ever reported Connected status.
+                    if (wasConnected && connectionStatus == ConnectionStatus.Disconnected)
+                    {
+                        s_deviceClient?.Dispose();
+                        wasConnected = false;
+                    }
+
+                    s_deviceClient = DeviceClient.CreateFromConnectionString(_deviceConnectionString, _transportType);
+                    s_deviceClient.SetConnectionStatusChangesHandler(ConnectionStatusChangeHandler);
+                    s_deviceClient.OperationTimeoutInMilliseconds = (uint)s_operationTimeout.TotalMilliseconds;
+                    _logger.LogDebug($"Reinitialized the client instance.");
+                }
+            }
+        }
+
+        private async Task SendMessagesAsync()
+        {
+            int count = 0;
+            while (true)
+            {
+                count++;
+                _logger.LogInformation($"Device sending message {count} to IoTHub...");
+
+                try
+                {
+                    await SendMessageAsync(count);
+                }
+                catch (Exception ex) when (ex is IotHubException exception && exception.IsTransient)
+                {
+                    // Inspect the exception to figure out if operation should be retried, or if user-input is required.
+                    _logger.LogError($"An IotHubException was caught, but will try to recover and retry explicitly: {ex}");
+                }
+                catch (Exception ex)
+                {
+                    if (ExceptionHelper.IsNetworkExceptionChain(ex))
+                    {
+                        _logger.LogError($"A network related exception was caught, but will try to recover and retry explicitly: {ex}");
+                    }
+                    else
+                    {
+                        throw;
+                    }
+                }
+
+                await Task.Delay((int)s_sleepDuration.TotalMilliseconds);
```

Comment:
Why not Task.Delay(s_sleepDuration)?

Reply:
no reason, I didn't know Task.delay could take in a TimeSpan directtly! I'll use that overload.